File: IntegrationCore/Models/DB/FieldDefinition.cs
namespace IntegrationCore.Models.DB
{
    public class FieldDefinition
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int TypeOfFieldId { get; set; }
        public bool IsArray { get; set; }
        public bool Required { get; set; }
        public string DefaultValue { get; set; }
        public int TypeId { get; set; }
        public TypeDefinition Type { get; set; }
    }
}

File: IntegrationCore/ClientApp/src/app/models/field-definition.ts
import { TypeDefinition } from './type-definition';

export class FieldDefinition {
  id: number;
  name: string;
  typeOfFieldId: number;
  isArray: boolean;
  isBasicType: boolean;
  required: boolean;
  defaultValue: string;
  typeId: number;
  type: TypeDefinition;
}
